### PR TITLE
DATAES-594 Added support for index without refresh.

### PR DESCRIPTION
**Pull request by murali-chevuri**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by murali-chevuri** [comment]
@sothawo Added junit test cases. Please review.

File: src/main/java/org/springframework/data/elasticsearch/repository/ElasticsearchRepository.java [hunk 1]
```diff
@@ -27,10 +27,13 @@
  * @author Rizwan Idrees
  * @author Mohsin Husen
  * @author Sascha Woo
+ * @author Murali Chevuri
  */
 @NoRepositoryBean
 public interface ElasticsearchRepository<T, ID> extends ElasticsearchCrudRepository<T, ID> {
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/repository/ElasticsearchRepository.java]: please add a Javadoc comment for this method that explicitly states that calling this method might lead to a temporary inconststent state until `ElasticsearchRepository.refrsh()`is called

**Comment by murali-chevuri** [comment]
> please add the mentioned comment and then squash your commits into one single commit with the commit message "_DATAES-594 Added support for index without refresh._". Then I'm ok with this.
Done the changes.

**Comment by sothawo** [comment]
The PR has been merged, thank you for your contribution.

**Pull request merged**